components/core/src/clp/CurlDownloadHandler.cpp

```diff
@@ -17,7 +20,17 @@ CurlDownloadHandler::CurlDownloadHandler(
         bool disable_caching,
         std::chrono::seconds connection_timeout,
         std::chrono::seconds overall_timeout
-) {
+)
+        : m_error_msg_buf{std::move(error_msg_buf)} {
+    if (nullptr != m_error_msg_buf) {
+        // Set up error message buffer
+        // According to the doc: https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html
+        // A successful call of setting `CURLOPT_ERRORBUFFER` will initialize the buffer to an empty
+        // string since 7.60.0. The minimum version we require is 7.68.0. Therefore, we don't need
+        // to manually clean up the passed-in buffer.
```

Comment:
```suggestion
        // According to the docs (https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html), since 7.60.0,
        // a successful call to set `CURLOPT_ERRORBUFFER` will initialize the buffer to an empty
        // string 7.60.0. Since we require at least 7.68.0, we don't need to clear the provided
        // buffer before it's used.
```